core/src/main/java/com/graphhopper/util/EdgeIteratorState.java

```diff
@@ -171,13 +171,21 @@ public void setBool(boolean reverse, IntsRef ref, boolean value) {
 
     EdgeIteratorState setReverse(DecimalEncodedValue property, double value);
 
-    <T extends Enum> T get(EnumEncodedValue<T> property);
+    <T extends Enum<?>> T get(EnumEncodedValue<T> property);
 
-    <T extends Enum> EdgeIteratorState set(EnumEncodedValue<T> property, T value);
+    <T extends Enum<?>> EdgeIteratorState set(EnumEncodedValue<T> property, T value);
 
-    <T extends Enum> T getReverse(EnumEncodedValue<T> property);
+    <T extends Enum<?>> T getReverse(EnumEncodedValue<T> property);
 
-    <T extends Enum> EdgeIteratorState setReverse(EnumEncodedValue<T> property, T value);
+    <T extends Enum<?>> EdgeIteratorState setReverse(EnumEncodedValue<T> property, T value);
+    
+    String get(StringEncodedValue property);
+    
+    EdgeIteratorState set(StringEncodedValue property, String value);
+    
+    String getReverse(StringEncodedValue property);
+    
+    EdgeIteratorState setReverse(StringEncodedValue property, String value);
```

Comment:
Here also the new method `set(property, fwd, bwd)` would be required now

Reply:
Rebased on master